[Zhou Jiahao] iP

zhoukerrr opened the pull request:
# DukePro

> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,

* text-based
* easy to learn
* ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2122s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

zhoukerrr committed:
Completed level-1

zhoukerrr committed:
Completed Level-2

zhoukerrr committed:
Completed Level-3

zhoukerrr committed:
Completed Level-4

zhoukerrr committed:
Completed A-TextUiTesting

zhoukerrr committed:
Updated bug in List

zhoukerrr committed:
Completed Level-5

zhoukerrr committed:
Completed Level-6

zhoukerrr committed:
Fix bugs

zhoukerrr committed:
Add Save Function

zhoukerrr committed:
Complete Level-7

zhoukerrr committed:
Complete Level-8

zhoukerrr committed:
Merge Level-7 and 8

zhoukerrr committed:
Load deadline date correctly

zhoukerrr committed:
Format output date correctly

zhoukerrr committed:
Refactor to more OOP

zhoukerrr committed:
Move to duke package

zhoukerrr committed:
Complete A-JUnit

zhoukerrr committed:
Add META-INF

zhoukerrr committed:
Add JavaDoc

zhoukerrr committed:
Follow coding standard

zhoukerrr committed:
Add find

zhoukerrr committed:
Merge code standard

zhoukerrr committed:
Merge JavaDoc

zhoukerrr committed:
Add more OOP

zhoukerrr committed:
Modify DukeTodoException message

zhoukerrr committed:
Merge branch 'brach-Level-8'

zhoukerrr committed:
Remove storage static method

zhoukerrr committed:
Update README.md

zhoukerrr committed:
Merge branch 'master' of https://github.com/Zhou-Jiahao-1998/ip

zhoukerrr committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

zhoukerrr committed:
Add Gradle

zhoukerrr committed:
Complete GUI

zhoukerrr committed:
Remove extra methods

zhoukerrr committed:
Update JavaDoc

zhoukerrr committed:
Merge level-10

JeffZincatz commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,53 @@
+import javafx.scene.image.ImageView;
+import javafx.scene.layout.HBox;
+
+public class DialogBox extends HBox {
```
Need JavaDoc header comment for public classes.

JeffZincatz commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,53 @@
+        this.getChildren().setAll(tmp);
+    }
+
+    public static DialogBox getUserDialog(Label l, ImageView iv) {
```
Need JavaDoc header comment for public methods.

JeffZincatz commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,55 @@
+     * @param input The command given by the user.
+     * @throws DukeException If input do not meet the requirements.
+     * @throws IOException If the file cannot be read/found.
+     * @returns The response in String after the action is executed.
```
An extra "s" after `@return` 😃

JeffZincatz commented on src/test/java/DukeTest.java:
```diff
@@ -0,0 +1,44 @@
+
+public class DukeTest {
+    @Test
+    public void todoNormalTest() {
```
Three part format `featureUnderTest_testScenario_expectedBehavior()` can be used for test method naming 😄

JeffZincatz reviewed (commented):
Very few coding style violations in general! Great job!

zhoukerrr committed:
Add Assertions

This is one of the requirement for the iP this week.

Assertions help to verify assumptions during runtime.

HalphasX commented on src/main/java/duke/DialogBox.java:
```diff
@@ -0,0 +1,53 @@
+        return new DialogBox(l, iv);
+    }
+
+    public static DialogBox getDukeDialog(Label l, ImageView iv) {
```
You have not added Javadoc comments to some of the methods. Otherwise, good job!

HalphasX commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,76 @@
+ */
+public class Task {
+    private String name;
+    private boolean done;
```
the Boolean naming convention for this can be changed to isDone.

HalphasX commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,76 @@
+    }
+
+    /**
+     * Mark the Task as done.
```
This should be "Marks" instead of "Mark" 😄

HalphasX reviewed (commented):
All in all, good functional code! Just a few nits that you should fix and LGTM!

zhoukerrr committed:
Refactor code

Followed the examples in notes.

I think this should be ok.

zhoukerrr committed:
Merge pull request #1 from Zhou-Jiahao-1998/branch-A-Assertions

Add Assertions

zhoukerrr committed:
Merge pull request #2 from Zhou-Jiahao-1998/branch-A-CodeQuality

Refactor code

zhoukerrr committed:
Refactor Parser

zhoukerrr committed:
Update style based on feedbacks

zhoukerrr committed:
Add help

zhoukerrr committed:
Add credit

zhoukerrr committed:
Add Readme

zhoukerrr committed:
Fix style

zhoukerrr committed:
Change font to fit all OS

zhoukerrr committed:
Update style
